specification/iotcentral/resource-manager/Microsoft.IoTCentral/preview/2021-11-01-preview/iotcentral.json

```diff
@@ -0,0 +1,1259 @@
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.IoTCentral/iotApps/{resourceName}": {
+      "get": {
+        "tags": [
+          "Apps"
+        ],
+        "description": "Get the metadata of an IoT Central application.",
+        "operationId": "Apps_Get",
+        "x-ms-examples": {
+          "Apps_Get": {
+            "$ref": "./examples/Apps_Get.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/resourceName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The body contains all the non-security properties of the IoT Central application. Security-related properties are set to null.",
+            "schema": {
+              "$ref": "#/definitions/App"
+            }
+          },
+          "default": {
+            "description": "Default error response",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            },
+            "x-ms-error-response": true
+          }
+        },
+        "deprecated": false
+      },
+      "put": {
+        "tags": [
+          "Apps"
+        ],
+        "description": "Create or update the metadata of an IoT Central application. The usual pattern to modify a property is to retrieve the IoT Central application metadata and security metadata, and then combine them with the modified values in a new body to update the IoT Central application.",
+        "operationId": "Apps_CreateOrUpdate",
+        "x-ms-examples": {
+          "Apps_CreateOrUpdate": {
+            "$ref": "./examples/Apps_CreateOrUpdate.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/resourceName"
+          },
+          {
+            "name": "App",
+            "in": "body",
+            "description": "The IoT Central application metadata and security metadata.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/App"
+            }
+          }
+        ],
+        "responses": {
```

Comment:
A PUT with a 200 will complete synchronously and no polling is necessary.
A PUT with a 201 will complete asynchronously and will return the polling information the response headers.

Return a 202 on a PUT is the old ARM model and was recently moved over to support 200/201

Here is the ARM RP contract for reference:
https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md